File: makeDelivery.rb
require 'uri'
require 'net/http'

url = URI("https://api.postmates.com/v1/customers/cus_KeDc-a3WqAD2rk/deliveries")

http = Net::HTTP.new(url.host, url.port)
http.use_ssl = true
http.verify_mode = OpenSSL::SSL::VERIFY_NONE

request = Net::HTTP::Post.new(url)
request["content-type"] = 'multipart/form-data; boundary=---011000010111000001101001'
request["authorization"] = 'Basic MGI1ODYyNjItNjQ4OS00Y2U4LTlhMzUtYmU4ZmIwZTc0NTE2Og=='
request["cache-control"] = 'no-cache'
request["postman-token"] = 'e307a6c0-64f5-8d26-261a-a201ece9aeaf'
request.body = "-----011000010111000001101001\r\nContent-Disposition: form-data; name=\"pickup_address\"\r\n\r\n76 9th Ave, New York, NY\r\n-----011000010111000001101001\r\nContent-Disposition: form-data; name=\"pickup_phone_number\"\r\n\r\n646-333-2271\r\n-----011000010111000001101001\r\nContent-Disposition: form-data; name=\"pickup_name\"\r\n\r\nTesting\r\n-----011000010111000001101001\r\nContent-Disposition: form-data; name=\"dropoff_address\"\r\n\r\nWorld Trade Center, New York, NY\r\n-----011000010111000001101001\r\nContent-Disposition: form-data; name=\"manifest\"\r\n\r\nTests\r\n-----011000010111000001101001\r\nContent-Disposition: form-data; name=\"dropoff_phone_number\"\r\n\r\n415-445-4444\r\n-----011000010111000001101001\r\nContent-Disposition: form-data; name=\"dropoff_name\"\r\n\r\nDeliverer\r\n-----011000010111000001101001\r\nContent-Disposition: form-data; name=\"quote_id\"\r\n\r\ndqt_KeE0IQusuIo1_k\r\n-----011000010111000001101001--"

response = http.request(request)
puts response.read_body

File: makedelivery.js


function getDeliveryQuote(pickup_address, dropoff_address){
	var data = new FormData();
	data.append("pickup_address", pickup_address);
	data.append("dropoff_address", dropoff_address);

	var xhr = new XMLHttpRequest();
	xhr.withCredentials = true;

	xhr.addEventListener("readystatechange", function () {
	  if (this.readyState === 4) {
	    console.log(this.responseText);
	  }
	});

	xhr.open("POST", "https://api.postmates.com/v1/customers/cus_KeDc-a3WqAD2rk/delivery_quotes");
	xhr.setRequestHeader("authorization", "Basic MGI1ODYyNjItNjQ4OS00Y2U4LTlhMzUtYmU4ZmIwZTc0NTE2Og==");
	xhr.setRequestHeader("cache-control", "no-cache");
	xhr.setRequestHeader("postman-token", "93863015-8f9c-6a42-dbde-4a9fc6cf1b79");

	xhr.send(data);

	return quote_id = JSON.parse(xhr.responseText).quote_id;
}

function createDelivery(pickup_address, pickup_name, pickup_phone_number, dropoff_address, dropoff_name, dropoff_phone_number, manifest, quote_id){
	var data = new FormData();
	data.append("pickup_address", pickup_address);				// company's address
	data.append("pickup_phone_number", pickup_phone_number);
	data.append("pickup_name", pickup_name);
	data.append("dropoff_address", dropoff_address);
	data.append("manifest", manifest);
	data.append("dropoff_phone_number", dropoff_phone_number);
	data.append("dropoff_name", dropoff_name);
	data.append("quote_id", quote_id);

	var xhr = new XMLHttpRequest();
	xhr.withCredentials = true;

	xhr.addEventListener("readystatechange", function () {
	  if (this.readyState === 4) {
	    console.log(this.responseText);
	  }
	});

	xhr.open("POST", "https://api.postmates.com/v1/customers/cus_KeDc-a3WqAD2rk/deliveries");
	xhr.setRequestHeader("authorization", "Basic MGI1ODYyNjItNjQ4OS00Y2U4LTlhMzUtYmU4ZmIwZTc0NTE2Og==");
	xhr.setRequestHeader("cache-control", "no-cache");
	xhr.setRequestHeader("postman-token", "86539afe-3218-40f4-aea0-da9e53ee6069");

	xhr.send(data);
}